File: backend/src/Alife.Domain/Entities/EventTravel.cs
using Alife.Domain.Enums;

namespace Alife.Domain.Entities;

/// <summary>Event-scoped driver evidence. Raw licence document content is deliberately not stored.</summary>
public sealed class EventTravelDriver
{
    public Guid Id { get; set; }
    public Guid EventId { get; set; }
    public Guid MemberId { get; set; }
    public string LicenceClass { get; set; } = string.Empty;
    public DateOnly? LicenceExpiresOn { get; set; }
    public bool LicenceConfirmed { get; set; }
    public bool FitToDriveConfirmed { get; set; }
    public string EvidenceNotes { get; set; } = string.Empty;
    public bool IsActive { get; set; } = true;
    public Guid VerifiedByMemberId { get; set; }
    public DateTime VerifiedUtc { get; set; }
    public Guid ConcurrencyToken { get; set; } = Guid.NewGuid();
    public DateTime CreatedUtc { get; set; }
    public DateTime UpdatedUtc { get; set; }

    public GroupEvent Event { get; set; } = null!;
    public Member Member { get; set; } = null!;
    public Member VerifiedByMember { get; set; } = null!;
    public ICollection<EventTravelJourney> Journeys { get; set; } = [];
}

/// <summary>Event-scoped vehicle evidence. It is not a public event projection.</summary>
public sealed class EventTravelVehicle
{
    public Guid Id { get; set; }
    public Guid EventId { get; set; }
    public string NameEn { get; set; } = string.Empty;
    public string NameZh { get; set; } = string.Empty;
    public string RegistrationReference { get; set; } = string.Empty;
    public int SeatCapacity { get; set; }
    public bool RegistrationConfirmed { get; set; }
    public DateOnly? RegistrationExpiresOn { get; set; }
    public bool WofConfirmed { get; set; }
    public DateOnly? WofExpiresOn { get; set; }
    public string EvidenceNotes { get; set; } = string.Empty;
    public bool IsActive { get; set; } = true;
    public Guid VerifiedByMemberId { get; set; }
    public DateTime VerifiedUtc { get; set; }
    public Guid ConcurrencyToken { get; set; } = Guid.NewGuid();
    public DateTime CreatedUtc { get; set; }
    public DateTime UpdatedUtc { get; set; }

    public GroupEvent Event { get; set; } = null!;
    public Member VerifiedByMember { get; set; } = null!;
    public ICollection<EventTravelJourney> Journeys { get; set; } = [];
}

public sealed class EventTravelJourney
{
    public Guid Id { get; set; }
    public Guid EventId { get; set; }
    public Guid EventOccurrenceId { get; set; }
    public string NameEn { get; set; } = string.Empty;
    public string NameZh { get; set; } = string.Empty;
    public DateTime StartUtc { get; set; }
    public DateTime EndUtc { get; set; }
    public Guid? DriverId { get; set; }
    public Guid? VehicleId { get; set; }
    public bool ManifestConfirmed { get; set; }
    public EventTravelJourneyStatus Status { get; set; } = EventTravelJourneyStatus.Planned;
    public Guid CreatedByMemberId { get; set; }
    public Guid ConcurrencyToken { get; set; } = Guid.NewGuid();
    public DateTime CreatedUtc { get; set; }
    public DateTime UpdatedUtc { get; set; }

    public GroupEvent Event { get; set; } = null!;
    public EventOccurrence EventOccurrence { get; set; } = null!;
    public EventTravelDriver? Driver { get; set; }
    public EventTravelVehicle? Vehicle { get; set; }
    public Member CreatedByMember { get; set; } = null!;
    public ICollection<EventTravelPickupStop> PickupStops { get; set; } = [];
    public ICollection<EventTravelPassengerAssignment> PassengerAssignments { get; set; } = [];
}

public sealed class EventTravelPickupStop
{
    public Guid Id { get; set; }
    public Guid JourneyId { get; set; }
    public int SortOrder { get; set; }
    public string NameEn { get; set; } = string.Empty;
    public string NameZh { get; set; } = string.Empty;
    public string AddressEn { get; set; } = string.Empty;
    public string AddressZh { get; set; } = string.Empty;
    public DateTime PickupUtc { get; set; }
    public DateTime CreatedUtc { get; set; }
    public DateTime UpdatedUtc { get; set; }

    public EventTravelJourney Journey { get; set; } = null!;
    public ICollection<EventTravelPassengerAssignment> PassengerAssignments { get; set; } = [];
}

/// <summary>Role-restricted manifest row. Ended rows are retained for operational audit.</summary>
public sealed class EventTravelPassengerAssignment
{
    public Guid Id { get; set; }
    public Guid JourneyId { get; set; }
    public Guid MemberId { get; set; }
    public Guid PickupStopId { get; set; }
    public Guid AssignedByMemberId { get; set; }
    public DateTime AssignedUtc { get; set; }
    public Guid? EndedByMemberId { get; set; }
    public DateTime? EndedUtc { get; set; }

    public EventTravelJourney Journey { get; set; } = null!;
    public Member Member { get; set; } = null!;
    public EventTravelPickupStop PickupStop { get; set; } = null!;
    public Member AssignedByMember { get; set; } = null!;
    public Member? EndedByMember { get; set; }
}

File: cloudflare/alife-app/src/types/eventTravel.ts
import type { LocalizedText } from './eventComposition'
import type { EventOccurrence } from './eventOperations'

export type EventTravelMemberOption = { memberId: string; displayName: string }

export type EventTravelDriver = {
  id: string
  eventId: string
  memberId: string
  memberDisplayName: string
  licenceClass: string
  licenceExpiresOn?: string | null
  licenceConfirmed: boolean
  fitToDriveConfirmed: boolean
  evidenceNotes: string
  isActive: boolean
  verifiedByMemberId: string
  verifiedUtc: string
  isEligible: boolean
  eTag: string
  createdUtc: string
  updatedUtc: string
}

export type SaveEventTravelDriverRequest = Omit<EventTravelDriver,
  'id' | 'eventId' | 'memberDisplayName' | 'verifiedByMemberId' | 'verifiedUtc' | 'isEligible' | 'eTag' | 'createdUtc' | 'updatedUtc'>

export type EventTravelVehicle = {
  id: string
  eventId: string
  name: LocalizedText
  registrationReference: string
  seatCapacity: number
  registrationConfirmed: boolean
  registrationExpiresOn?: string | null
  wofConfirmed: boolean
  wofExpiresOn?: string | null
  evidenceNotes: string
  isActive: boolean
  verifiedByMemberId: string
  verifiedUtc: string
  evidenceComplete: boolean
  eTag: string
  createdUtc: string
  updatedUtc: string
}

export type SaveEventTravelVehicleRequest = Omit<EventTravelVehicle,
  'id' | 'eventId' | 'verifiedByMemberId' | 'verifiedUtc' | 'evidenceComplete' | 'eTag' | 'createdUtc' | 'updatedUtc'>

export type EventTravelPickupStop = {
  id: string
  journeyId: string
  sortOrder: number
  name: LocalizedText
  address: LocalizedText
  pickupUtc: string
}

export type SaveEventTravelPickupStopRequest = Omit<EventTravelPickupStop, 'id' | 'journeyId'>

export type EventTravelPassenger = {
  id: string
  journeyId: string
  memberId: string
  memberDisplayName: string
  pickupStopId: string
  pickupStopName: LocalizedText
  pickupUtc: string
  assignedByMemberId: string
  assignedUtc: string
}

export type EventTravelJourneyStatus = 'planned' | 'confirmed' | 'cancelled'

export type EventTravelJourney = {
  id: string
  eventId: string
  eventOccurrenceId: string
  name: LocalizedText
  startUtc: string
  endUtc: string
  driver?: EventTravelDriver | null
  vehicle?: EventTravelVehicle | null
  pickupStops: EventTravelPickupStop[]
  passengerManifest: EventTravelPassenger[]
  passengerCount: number
  manifestConfirmed: boolean
  status: EventTravelJourneyStatus
  eTag: string
  createdUtc: string
  updatedUtc: string
}

export type EventTravelRamEvidence = {
  transportRequired?: boolean | null
  licensedDriverConfirmed?: boolean | null
  vehicleRegistrationConfirmed?: boolean | null
  vehicleWofConfirmed?: boolean | null
  status: 'draft' | 'awaitingReview' | 'approved'
  checksComplete: boolean
}

export type EventTravelReadiness = {
  transportFactsConfirmed: boolean
  driversAndVehiclesQualified: boolean
  passengerManifestsComplete: boolean
  ramTransportChecksComplete: boolean
  blockers: LocalizedText[]
}

export type EventTravelWorkspace = {
  eventId: string
  occurrences: EventOccurrence[]
  eligibleMembers: EventTravelMemberOption[]
  drivers: EventTravelDriver[]
  vehicles: EventTravelVehicle[]
  journeys: EventTravelJourney[]
  ramEvidence: EventTravelRamEvidence
  readiness: EventTravelReadiness
  canManage: boolean
  dataClassification: 'roleRestricted'
}

export type CreateEventTravelJourneyRequest = {
  eventOccurrenceId: string
  name: LocalizedText
  startUtc: string
  endUtc: string
  driverId?: string | null
  vehicleId?: string | null
}

export type UpdateEventTravelJourneyRequest = Omit<CreateEventTravelJourneyRequest, 'eventOccurrenceId'> & {
  manifestConfirmed: boolean
  status: EventTravelJourneyStatus
}

export type EventTravelMyJourney = {
  journeyId: string
  eventOccurrenceId: string
  name: LocalizedText
  startUtc: string
  endUtc: string
  driverDisplayName?: string | null
  vehicleName?: LocalizedText | null
  vehicleRegistrationReference?: string | null
  pickupStopName?: LocalizedText | null
  pickupStopAddress?: LocalizedText | null
  pickupUtc?: string | null
  status: EventTravelJourneyStatus
}

export type EventTravelMyJourneys = {
  eventId: string
  journeys: EventTravelMyJourney[]
  dataClassification: 'userSpecific'
}
